src/lib/log.c

```diff
@@ -20,50 +20,88 @@
 #define _GNU_SOURCE
 
 #include <ctype.h>
+#include <limits.h>
 #include <stdarg.h>
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
 
 #include "log.h"
+
 #define DEBUG_HEADER_SIZE 50
+#define LINE_MAX _POSIX2_LINE_MAX
 
 static void log_internal(FILE *out, const char *file, int line, const char *label, const char *format, va_list args_list);
 static int cur_log_level = LOG_INFO;
 static log_fn_t log_function = log_internal;
 
-static void log_internal(FILE *out, const char *file, int line, const char *label, const char *format, va_list args_list)
+static void print_debug_info(FILE *out, const char *file, int line)
 {
 	char time_str[50];
 	struct tm *timeinfo;
 	int printed;
+	time_t currtime;
+
+	time(&currtime);
+	timeinfo = localtime(&currtime);
+	strftime(time_str, sizeof(time_str), "%Y-%m-%d %H:%M:%S", timeinfo);
+	printed = fprintf(out, "%s (%s:%d)", time_str, file, line);
+	if (printed < DEBUG_HEADER_SIZE) {
+		fprintf(out, "%*c", 50 - printed, ' ');
+	}
+}
+
+static void log_internal(FILE *out, const char *file, int line, const char *label, const char *format, va_list args_list)
+{
+
+	char msg[LINE_MAX];
+
+	/* do not print the carriage return found at the beginning of the message if
+	 * in debug mode or if printing to a file, that would mess up the output */
+	while (strncmp(format, "\r", 1) == 0) {
+		if (cur_log_level != LOG_DEBUG && (out == stdout || out == stderr)) {
```

Comment:
I just test this again and this isn't working. '\r' are being outputted to a file. If you redirect the output to a file out is still going to be stdout or stderr. In progress we check if we are outputting to a tty using isatty().
I still prefer to remove this code and handle that only on progress. But as I said I'm ok to leave it here if you prefer. But check for tty instead.

Reply:
you are right, I will change this